ledger/error/src/main/scala/com/daml/error/definitions/groups/PartyManagementServiceErrorGroup.scala

```diff
@@ -0,0 +1,152 @@
+        ErrorCategory.ContentionOnSharedResources,
+      ) {
+    case class Reject(party: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends DamlError(
+          cause =
+            s"Update operation for party '$party' failed due a concurrent update to the same party"
+        ) {
+      override def resources: Seq[(ErrorResource, String)] = Seq(
+        ErrorResource.Party -> party
+      )
+    }
+  }
+
+  @Explanation("The party referred to by the request was not found.")
+  @Resolution(
+    "Check that you are connecting to the right participant node and the party is spelled correctly."
+  )
+  object PartyNotFound
+      extends ErrorCode(
+        id = "PARTY_NOT_FOUND",
+        ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
+      ) {
+    case class Reject(operation: String, party: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends DamlErrorWithDefiniteAnswer(
+          cause = s"$operation failed for unknown party '$party'"
+        ) {
+      override def resources: Seq[(ErrorResource, String)] = Seq(
+        ErrorResource.Party -> party
+      )
+    }
+  }
+
+  @Explanation(
+    """|Each on-ledger party known to this participant node can a have a participant local metadata assigned to it.
+               |The participant local information about a party referred to by this request was not found while it should have been found."""
+  )
+  @Resolution(
+    "This error can indicate a problem with the server's storage or implementation."
+  )
+  object InternalPartyRecordNotFound
+      extends ErrorCode(
+        id = "INTERNAL_PARTY_RECORD_NOT_FOUND",
+        ErrorCategory.SystemInternalAssumptionViolated,
+      ) {
+    case class Reject(operation: String, party: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends DamlErrorWithDefiniteAnswer(
+          cause = s"$operation failed as party record identified by party '$party' was not found"
+        ) {
+      override def resources: Seq[(ErrorResource, String)] = Seq(
+        ErrorResource.Party -> party
+      )
+    }
+  }
+
+  @Explanation(
+    """|Each on-ledger party known to this participant node can a have a participant local metadata assigned to it.
+                  |The participant local information about a party referred to by this request was found while it should have been not found."""
+  )
+  @Resolution(
+    "This error can indicate a problem with the server's storage or implementation."
+  )
+  object InternalPartyRecordAlreadyExists
+      extends ErrorCode(
+        id = "INTERNAL_PARTY_RECORD_ALREADY_EXISTS",
+        ErrorCategory.SystemInternalAssumptionViolated,
+      ) {
+    case class Reject(operation: String, party: String)(implicit
+        loggingContext: ContextualizedErrorLogger
+    ) extends DamlErrorWithDefiniteAnswer(
+          cause = s"$operation failed as party record identified by party '$party' already exists"
```

Comment:
```suggestion
          cause = s"Party record for '$party' already exists in $operation"
```

Reply:
Changing to `s"Party record for party: '$party' already exists when $operation"`